### PR TITLE
SET DATABASE docs

### LINKED ISSUE #451: Encourage use of db connection string instead of SET DATABASE
@mjibson, from Gitter:

`SET DATABASE` is dangerous because Go sometimes makes a new connection. Use the path of a connection string to specify the DB.

So:
- On `SET DATABASE` doc, call this out and encourage setting the db in the connection string. 
- Make sure we're not encouraging `SET DATABASE` throughout our SQL reference docs.

### LINKED ISSUE #495: SET DATABASE statement
Intro, Synopsis, and Required Privileges are complete. We need to complete **Parameters** and **Examples**. For style reference, see those sections in [`INSERT`](insert.html) or [`DELETE`](delete.html).

### PR DESCRIPTION
**Pull request by jseldess**
This PR completes our docs on `SET DATABASE`. 

Fixes #451 
Fixes #495

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/671)

### CODE REVIEW

**Comment by jseldess** [comment]
@mjibson, you'll notice that I simplified the railroad diagram, but I couldn't figure out how to get it down to one main line. Let me know if you have any ideas.

HTML version: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/5c24191ea3348e485c8402f7dcf0704cd4816083/set-database.html

**Comment by jseldess** [comment]
@mjibson, @sploiselle and I figure out the diagram.

File: generate/main.go [hunk 1]
```diff
@@ -208,6 +208,7 @@ func main() {
 				{name: "savepoint_stmt", inline: []string{"savepoint_name"}},
 				{name: "select_stmt", inline: []string{"select_no_parens", "simple_select", "opt_sort_clause", "select_limit"}},
 				{name: "set_stmt", inline: []string{"set_rest", "set_rest_more", "generic_set"}, exclude: regexp.MustCompile("CHARACTERISTICS"), replace: map[string]string{"'TRANSACTION' transaction_mode_list | ": ""}},
+				{name: "set_database", stmt: "set_stmt", inline: []string{"set_rest", "set_rest_more", "generic_set"}, exclude: regexp.MustCompile("CHARACTERISTICS"), replace: map[string]string{"var_name": "'DATABASE'", "'DEFAULT'": "var_list", "var_list": "database_name", "'LOCAL'": "", "'SESSION'": "", "'TRANSACTION' transaction_mode_list | ": ""," | 'TIME' 'ZONE' zone_value ": ""},unlink: []string{"database_name"}},
```
Comment (mjibson) [inline generate/main.go]: I think with a `match` you can get rid of the optional empty branch.

Comment (jseldess) [inline generate/main.go]: @mjibson, we figured it out using more replacements, but I forgot to update the link. You can see it here: http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/dfafce3078cf944c7eb72fe42f9125f24e8691c4/set-database.html

I believe I tried with `match` and couldn't get it to work. If you have a sec to help out, I'm happy to try again, if you like.

**Pull request merged**